multicluster/service-mirror/cluster_watcher.go

```diff
@@ -1084,83 +1083,6 @@ func (rcsw *RemoteClusterServiceWatcher) handleCreateOrUpdateEndpoints(
 	return err
 }
 
-// createEndpointMirrorService creates a new Endpoint Mirror service and its
-// corresponding endpoints object. It returns the newly created Endpoint Mirror
-// service object. When a headless service is exported, we create a Headless
-// Mirror service in the source cluster and then for each hostname in the
-// exported service's endpoints object, we also create an Endpoint Mirror
-// service (and its corresponding endpoints object).
-func (rcsw *RemoteClusterServiceWatcher) createEndpointMirrorService(ctx context.Context, endpointHostname, resourceVersion, endpointMirrorName string, exportedService *corev1.Service) (*corev1.Service, error) {
```

Comment:
This is only used by headless mirrors so it should be moved to `cluster_watcher_headless.go`, similar to the rest of that code.